oggsound/src/OgreOggSoundManager.cpp

```diff
@@ -2501,26 +2499,32 @@ namespace OgreOggSound
 	/*/////////////////////////////////////////////////////////////////*/
 	int OgreOggSoundManager::_createSourcePool()
 	{
-		ALuint source;
-		unsigned int numSources = 0;
+		ALuint sources[mMaxSources];
+
+		mSources = new ALuint[mMaxSources];
+
+		alGenSources(mMaxSources, mSources);
 
-		while(alGetError() == AL_NO_ERROR && numSources < mMaxSources)
+		ALenum error = 0;
+		if ((error = alGetError()) != AL_NO_ERROR)
 		{
-			source = 0;
-			alGenSources(1,&source);
-			if(source != 0)
-			{
-				mSourcePool.push_back(source);
-				numSources++;
-			}
-			else
+			switch (error)
 			{
-				alGetError();
-				break;
+			case AL_INVALID_VALUE:		{ LogManager::getSingleton().logMessage("Invalid Value when attempting to create OpenAL sources", Ogre::LML_CRITICAL);}		break;
```

Comment:
```suggestion
			case AL_INVALID_VALUE:		{ LogManager::getSingleton().logError("Invalid Value when attempting to create OpenAL sources");}		break;
```

Reply:
@paroj let's leave this change for a future commit with all the `logMessages()` replaced by `logError()` (when appropiate) there are lots of places where this replacement is needed.